include/dlaf/matrix/matrix_base.h

```diff
@@ -69,6 +75,10 @@ class MatrixBase {
   MatrixBase(MatrixBase&& rhs) = default;
   MatrixBase& operator=(MatrixBase&& rhs) = default;
 
+  static std::size_t futureVectorSize(const LocalTileSize& local_nr_tiles) noexcept {
```

Comment:
`MatrixBase` is an implementation detail which is used to avoid code duplication among
`Matrix` and `MatrixView`.
Moreover this method is a protected method used internally to return the size needed for the future vectors which **currently** (no guarantee that we will not change that) correspond to the number of tiles.